### PR TITLE
Use tox for tests

### LINKED ISSUE #224: Explicit python version in Makefile
At the moment we use `python3` as our python version in Makefile
https://github.com/wellcometrust/WellcomeML/blob/9acf7cea9641000980e5bb4bfe80daebc5c5e5c9/Makefile#L4

This is not reproducible as each of us including travis might be running on a different python. We force travis to try python 3.7 and 3.8 which is a good thing but we should also ensure our env is created with one version.

An added benefit of this is that brew cannot mess up your python as python3.8 should link to one version compared to python3 that might be updated to point to the latest.

### PR DESCRIPTION
**Pull request by nsorros**
Description
---

Fixes #224

Checklist
---

- [x] Added link to Github issue or Trello card
- [x] Added tests

### CODE REVIEW

**Comment by nsorros** [comment]
[Switch WellcomeML to tox](https://trello.com/c/Muz3FagU/1191-switch-wellcomeml-to-tox)

**Comment by nsorros** [comment]
As part of this PR we could potentially switch to Github actions, opened an issue for discussion #247. @lizgzil @aCampello

**Comment by nsorros** [comment]
New tool, new problems. Python3.7 with tox complains about binary incompatibility when installing spacy models. To make things more interestingly, it runs fine locally, a problem that tox should not have as it runs tests in its own isolated environment which is more reproducible than what we were doing before.

File: Makefile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 .DEFAULT_GOAL := all
 
 VIRTUALENV := build/virtualenv
-PYTHON_VERSION := python3
+PYTHON_VERSION := python3.8
```
Comment (aCampello) [inline Makefile:4]: Any reason why our default is now 3.8? (I'm fine with that btw)

File: tox.ini [hunk 2]
```diff
@@ -0,0 +1,11 @@
+[tox]
+envlist = py37, py38
+
+[testenv]
+deps = 
+	-r requirements_test.txt
+	.[deep-learning]
+
+commands = python -m spacy download en_core_web_sm
+		   pytest -m  "not (integration)" -s -v --durations=0 --disable-warnings --tb=line --cov=wellcomeml ./tests
```
Comment (aCampello) [inline tox.ini:10]: is there an indentation problem here?

**Review by aCampello — APPROVED** [review]
Thanks for this change! Looks good!

File: Makefile (hunk 1 above)
Comment (nsorros) [reply Makefile:4]: Check #224 and #228

File: tox.ini (hunk 2 above)
Comment (nsorros) [reply tox.ini:10]: ini files do not care about indentation so I guess that is why this works but let me see whether this is a space vs tabs issue

File: Makefile (hunk 1 above)
Comment (aCampello) [reply Makefile:4]: Lol - it was my suggestion to keep 3.8.... 🙈

File: tox.ini (hunk 2 above)
Comment (nsorros) [reply tox.ini]: 7888bd292e74f569703ec428a5485c87b1ca4860

**Comment by nsorros** [comment]
On the way we seem to have missed the codecov report/ @aCampello @lizgzil. Investigating.

**Comment by nsorros** [comment]
Open a separate issue for the codecov problem #251

**Comment by nsorros** [comment]
Btw @lizgzil and @aCampello note, that to run tests moving on you need to run `tox` and you need need `python 3.7` and `python 3.8`. If one is missing tests will not run for that version. More importantly, the default python version for development has been switched to `python 3.8` as discussed in #224. If you have any problem installing that version let me know. I have been using `pyenv` to manage multiple python versions which has been working fine.

**Pull request merged**